source/common/upstream/load_balancer_impl.cc

```diff
@@ -67,8 +68,17 @@ void LoadBalancerBase::recalculatePerPriorityState(uint32_t priority) {
                                  host_set.healthyHosts().size() / host_set.hosts().size()));
   }
 
-  // Now that we've updated health for the changed priority level, we need to caculate percentage
+  // Now that we've updated health for the changed priority level, we need to calculate percentage
   // load for all priority levels.
+
+  // The following cases are handled here;
+  // - Total health is = 100. It means there are enough healthy hosts to handle the load.
+  //   Do not enter panic mode, even if a specific priority has low number of healthy hosts.
+  // - Total health is < 100. There is not enough healthy hosts to handle the load. Continue
+  //   distibuting the load among priority sets, but turn on Panic mode if # of healthy hosts
+  //   in priority set is low.
+  // - Total health is 0. All hosts are down. Redirect 100% of traffic to P=0 and enable PanicMode.
```

Comment:
super nitty nit but I see panic mode, Panic mode, and PanicMode in this comment - mind picking one and updating the comments in this PR?